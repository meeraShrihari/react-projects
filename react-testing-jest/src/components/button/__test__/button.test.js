// import { render, screen } from '@testing-library/react';
// import Button from "../button";

// test("renders without crashing", () => {
//     render(<Button />);
//     const buttonElement = screen.getByTestId("button");
//     expect(buttonElement).toBeInTheDocument();
// });

import {render, screen, cleanup} from "@testing-library/react";
import Button from "../button";
import "@testing-library/jest-dom"
import renderer from "react-test-renderer";

afterEach(cleanup);

test("Renders button on the dom", () => {
    render(<Button />);
    const buttonElement = screen.getByTestId("button");
    expect(buttonElement).toBeInTheDocument();
})

test("Renders button correctly", () => {
    render(<Button label="Click me please"></Button>);
    const buttonElement = screen.getByTestId("button");
    expect(buttonElement).toHaveTextContent("Click me please");
})

// test("Renders button with save", () => {
//     render(<Button label="save" />);
//     const buttonElement = screen.getByTestId("button");
//     expect(buttonElement).toHaveTextContent("save");
// })


test("matches snapshot", () => {
    const component = renderer.create(<Button label="click me please"></Button>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test("matches snapshot2", () => {
    const component = renderer.create(<Button label="save"></Button>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})