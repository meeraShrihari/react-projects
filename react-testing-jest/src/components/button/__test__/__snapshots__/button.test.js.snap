// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi 
  click me please
</div>
`;

exports[`matches snapshot2 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi 
  save
</div>
`;
